File: backend/app/schemas/artifact.py
from datetime import datetime
from typing import Optional, Dict, Any, List
from pydantic import BaseModel, ConfigDict, Field


class ArtifactCreate(BaseModel):
    title: str = "Untitled Artifact"
    type: str  # markdown, html
    content: str
    message_id: Optional[str] = None


class ArtifactResponse(BaseModel):
    id: str
    session_id: str
    message_id: Optional[str] = None
    title: str
    type: str
    content: str
    metadata: Dict[str, Any] = Field(default_factory=dict, alias="artifact_metadata")
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True, populate_by_name=True)


class ArtifactListResponse(BaseModel):
    artifacts: List[ArtifactResponse]

File: frontend/src/types/artifact.ts
export type ArtifactType = 'markdown' | 'html';

export interface Artifact {
  id: string;
  session_id: string;
  message_id?: string | null;
  title: string;
  type: ArtifactType;
  content: string;
  metadata?: Record<string, any>;
  created_at: string;
  updated_at: string;
}

export interface ArtifactListResponse {
  artifacts: Artifact[];
}

export interface ArtifactCreatePayload {
  title?: string;
  type: ArtifactType;
  content: string;
  message_id?: string;
}
